### ISSUE TITLE
Goal 'upload-bom' does not match findings against securityGate under some conditions since version 2.4.0

### ISSUE THREAD

**Issue by dollmmino** [iabudiab/dependency-track-maven-plugin#19]
Hello,

**Environment**
Our pom contains the following:
```
...
<plugin>
    <groupId>dev.iabudiab</groupId>
    <artifactId>dependency-track-maven-plugin</artifactId>
    <version>2.4.0</version>
    <configuration>
        <dependencyTrackUrl>SOME_URL</dependencyTrackUrl>
        <projectName>SOME_NAME</projectName
        </securityGate>
            <critical>0</critical>
            <high>0</high>
            <medium>0</medium>
            <low>0</low>
        <securityGate>
    </configuration>
</plugin>
...
```

**Steps to reproduce**
1. Setup a project including a pom.xml containing the part in Environment.
2. Add a dependency to the pom, that has a known vulnerability
3. Create a CycloneDX Bom, e.g using org.cyclonedx:cyclonedx-maven-plugin and executing the following:
`mvn org.cyclonedx:cyclonedx-maven-plugin:makeAggregateBom`
4. Execute the 'upload-bom' goal:
`mvn --projects . dev.iabudiab:dependency-track-maven-plugin:upload-bom -DskipTests -DdependencyTrackApiKey=SOME_API_KEY`

**Expected Result**
I expect that the upload-bom goal matches the findings against the configured securityGate. I added a dependency to the pom that has a MEDIUM vulnerability:
```
...
[INFO] Upload Matching suppressions    False
[INFO] Reset expired suppressions      true
[INFO] ProjectMetrics retry delay      5
[INFO] ProjectMetrics retry limit      3
[INFO] Using API v1 at                 SOME_URL
[INFO] Uploading Bom artifact to       SOME_URL
[INFO] BOM response token:             SOME_TOKEN
[INFO] Token has been written to       SOME_PATH
[INFO] Polling Token                   SOME_TOKEN
[INFO] Token is still beeing processed, will retry in 5 seconds
[INFO] Polling token:                  SOME_TOKEN
[INFO] --- Findings report ---
...
[INFO] --- Project Metrics ---
...
- Critical: 0
- High    : 0
- Medium  : 1
- Low     : 0
[INFO] --- Security Gate ---
...
    critical: 0, high: 0, medium: 0, low: 0
[INFO] --- Custom Suppressions ---
    None
[INFO] Skip checking for matching suppressions to be uploaded
[INFO] --- Report ---
- Active Finding for ...
...
[WARNING] Project did not pass the security gate
[INFO] ------------------------------------------------------
[INFO] BUILD FAILURE
...
```

**Actual Result**
No match of findings against security gate executed.
```
...
[INFO] Upload Matching suppressions    False
[INFO] Reset expired suppressions      true
[INFO] ProjectMetrics retry delay      5
[INFO] ProjectMetrics retry limit      3
[INFO] Using API v1 at                 SOME_URL
[INFO] Uploading Bom artifact to       SOME_URL
[INFO] BOM response token:             SOME_TOKEN
[INFO] Token has been written to       SOME_PATH
[INFO] Polling Token                   SOME_TOKEN
[INFO] Token is still beeing processed, will retry in 5 seconds
[INFO] Polling token:                  SOME_TOKEN
[INFO] --- Findings report ---
...
[INFO] --- Project Metrics ---
...
- Critical: 0
- High    : 0
- Medium  : 1
- Low     : 0
[INFO] --- Security Gate ---
...
    critical: 0, high: 0, medium: 0, low: 0
[INFO] --- Custom Suppressions ---
    None
[INFO] Skip checking for matching suppressions to be uploaded
[INFO] ------------------------------------------------------
[INFO] BUILD SUCCESS
...
```

**Possible Issue**
This issue doesn't occur when using version 2.3.0.

I looked into the commits and the bug might have happened in the commit f35ac4a 'Refactor code base for a better separation of concerns'.

In the old version of UploadBomMojo.java's doWork() Method there is a call line 207:
`SecurityGate.SecurityReport securityReport = securityGate.applyOn(findings, suppressions)`
I assume this line matches the findings against the securityGate, and I also think that this line might not be executed in the new version.

The reason for that might be this if-case:
```
if (findings == null || !uploadMatchingSuppressions) {
    getLog().info("Skip checking for matching suppressions to be uploaded");
    return;
}
```
in the old version this if-case is part of the method `uploadSuppressions` that is called right above the `securityGate.applyOn` call in line 205.
In our configuration (see logs and configuration above) we set the `uploadMatchingSuppressions` property to false. So in our case we will go into that if-case.
Now, in the old version that means that the method `uploadSuppressions` is just returning early. But in the new version this if-case is part of the doWork() method and is above the `securityGate.applyOn` call. So this if-case might early return the `doWork()` method in the new version, preventing the matching to be called.
Therefore in the `actual result` you might also see the log `Skip checking for matching suppressions to be uploaded` as the last thing that happens during the plugins goal execution.

If that is right, you would need to set the `uploadMatchingSuppressions` property to true in order to execute the `securityGate.applyOn`. Which is a different behavior than before, e.g. in version 2.3.0 there you did not need to set this property to true.

**Comment by iabudiab** [comment]
@dollmmino Hey there 👋 thank you vey much for this very detailed bug report 💎 

I'll check, fix, and release an update as soon as possible.

**Comment by iabudiab** [comment]
@dollmmino Thanks again for the issue. It's been fixed now. I've released the new version (2.4.1) and uploaded it to maven central. I think, it would be available sometime tomorrow.
